### PR TITLE
fix: [CDS-41906]: service now approval - refactor and fix calling create meta api twice, remove unnecessary useEffect

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

- refactor and remove unnecessary useEffect, useState whose values can be derived
- fix calling create meta API twice

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx [hunk 1]
```diff
@@ -71,25 +66,15 @@ function FormContent({
...
   allowableTypes,
   stepViewType,
-  refetchServiceNowTicketTypes,
-  fetchingServiceNowTicketTypes,
-  serviceNowTicketTypesResponse,
-  serviceNowTicketTypesFetchError,
-  refetchServiceNowMetadata,
-  serviceNowMetadataResponse,
-  fetchingServiceNowMetadata,
-  serviceNowMetadataFetchError
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx:81]: passing 8 props is cumbersome and just pass the query.

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 9 Issues 
* Code issue 9 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10795)

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx [hunk 2]
```diff
@@ -105,15 +90,16 @@ function FormContent({
       ? formik.values.spec.ticketType
       : undefined
 
-  useEffect(() => {
-    if (ticketTypeKeyFixedValue && serviceNowMetadataResponse?.data) {
-      setTicketFieldList(serviceNowMetadataResponse?.data)
-    }
-  }, [serviceNowMetadataResponse?.data, ticketTypeKeyFixedValue])
+  const serviceNowTicketTypesOptions: ServiceNowTicketTypeSelectOption[] =
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx:93]: unnecessary useEffect and useState whose value can be derived. 

We may not need a further wrapper with useMemo for the derided value. Rather avoid unnecessary rerenders

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx [hunk 3]
```diff
@@ -136,22 +122,10 @@ function FormContent({
       const rejectionCriteria = getApprovalRejectionCriteriaForInitialValues(formik.values.spec.rejectionCriteria)
       formik.setFieldValue('spec.rejectionCriteria', rejectionCriteria)
     }
-  }, [serviceNowTicketTypesOptions, ticketTypeKeyFixedValue])
-
-  useEffect(() => {
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx:141]: unnecessary useEffect

**Review by rameshx — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Pull request merged**